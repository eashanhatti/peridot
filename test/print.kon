let {
	lazy print_xy : (IO Unit) =
		(bind
			(print x)
			\ _ { (bind
				(print y)
				\ _ { (pure unit) })});
} in {
	print_xy
}
