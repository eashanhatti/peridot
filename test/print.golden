((fromList
    [(Global {unGlobal = 0}, Just UnitType),
     (Global {unGlobal = 1}, Just UnitType),
     (Global {unGlobal = 2},
      Just (IOType (UniVar (Global {unGlobal = 0})))),
     (Global {unGlobal = 3}, Nothing),
     (Global {unGlobal = 4}, Just UnitType),
     (Global {unGlobal = 5},
      Just (IOType (UniVar (Global {unGlobal = 3})))),
     (Global {unGlobal = 6}, Just UnitType)],
  fromList [],
  fromList [],
  []),
 Let
   [ObjTerm
      (Id {unId = 2})
      (IOType UnitType)
      (IOIntroBind
         (PutChar 'x')
         (ObjectFunIntro
            (IOIntroBind
               (PutChar 'y')
               (ObjectFunIntro (IOIntroPure UnitIntro)))))]
   (GlobalVar (Id {unId = 2})))
